api/appStore/deployment/AppStoreDeploymentNewRestHandler.go

```diff
@@ -0,0 +1,263 @@
+
+	isChartRepoActive, err := handler.appStoreDeploymentService.IsChartRepoActive(request.AppStoreVersion)
+	if err != nil {
+		handler.Logger.Errorw("service err, CreateInstalledApp", "err", err, "payload", request)
+		common.WriteJsonResp(w, err, nil, http.StatusInternalServerError)
+		return
+	}
+	if !isChartRepoActive {
+		common.WriteJsonResp(w, fmt.Errorf("chart repo is disabled"), nil, http.StatusNotAcceptable)
+		return
+	}
+
+	request.UserId = userId
+	handler.Logger.Infow("request payload, CreateInstalledApp", "payload", request)
+	ctx, cancel := context.WithCancel(r.Context())
+	if cn, ok := w.(http.CloseNotifier); ok {
+		go func(done <-chan struct{}, closed <-chan bool) {
+			select {
+			case <-done:
+			case <-closed:
+				cancel()
+			}
+		}(ctx.Done(), cn.CloseNotify())
+	}
+	ctx = context.WithValue(r.Context(), "token", token)
+	defer cancel()
+	res, err := handler.appStoreDeploymentService.InstallApp(&request, ctx)
+	if err != nil {
+		if strings.Contains(err.Error(), "application spec is invalid") {
+			err = &util.ApiError{Code: "400", HttpStatusCode: 400, UserMessage: "application spec is invalid, please check provided chart values"}
+		}
+		handler.Logger.Errorw("service err, CreateInstalledApp", "err", err, "payload", request)
+		common.WriteJsonResp(w, err, nil, http.StatusInternalServerError)
+		return
+	}
+	common.WriteJsonResp(w, err, res, http.StatusOK)
+}
+
+func (handler AppStoreDeploymentRestHandlerImpl) GetInstalledAppsByAppStoreId(w http.ResponseWriter, r *http.Request) {
+	userId, err := handler.userAuthService.GetLoggedInUser(r)
+	if userId == 0 || err != nil {
+		common.WriteJsonResp(w, err, "Unauthorized User", http.StatusUnauthorized)
+		return
+	}
+
+	vars := mux.Vars(r)
+	appStoreId, err := strconv.Atoi(vars["appStoreId"])
+	if err != nil {
+		handler.Logger.Errorw("request err, GetInstalledAppsByAppStoreId", "err", err, "appStoreId", appStoreId)
+		common.WriteJsonResp(w, err, nil, http.StatusBadRequest)
+		return
+	}
+	token := r.Header.Get("token")
+	handler.Logger.Infow("request payload, GetInstalledAppsByAppStoreId", "appStoreId", appStoreId)
+	res, err := handler.appStoreDeploymentService.GetAllInstalledAppsByAppStoreId(w, r, token, appStoreId)
+	if err != nil {
+		handler.Logger.Errorw("service err, GetInstalledAppsByAppStoreId", "err", err, "appStoreId", appStoreId)
+		common.WriteJsonResp(w, err, nil, http.StatusInternalServerError)
+		return
+	}
+
+	var installedAppsResponse []appStoreBean.InstalledAppsResponse
+	for _, app := range res {
+
+		//rbac block starts from here
+		var rbacObject string
+		if app.AppOfferingMode == util2.SERVER_MODE_HYPERION {
+			rbacObject = handler.enforcerUtilHelm.GetHelmObjectByClusterId(app.ClusterId, app.Namespace, app.AppName)
+		}else{
+			rbacObject = handler.enforcerUtil.GetHelmObjectByAppNameAndEnvId(app.AppName, app.EnvironmentId)
+		}
+		if ok := handler.enforcer.Enforce(token, casbin.ResourceHelmApp, casbin.ActionGet, rbacObject); !ok {
+			continue
```

Comment:
we should apply authz before fetching data from ACD

Reply:
ACD means - getStatus?

Reply:
this logic/code is legacy for argocd. so avoiding to do as for now